src/ApplicationInsights.Kubernetes/TelemetryInitializers/TelemetryKeyCache.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Concurrent;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.ApplicationInsights.Kubernetes
+{
+    internal class TelemetryKeyCache : ITelemetryKeyCache
+    {
+        // The capacity of 40 is kind of arbitrary. We have around 10 keys appended to for the telemetry;
+        // Times that by 4 to allow a bit of wiggling room for name mapping;
+        // However, having too many of the mapped key is not allowed to prevent huge memory consuption.
+        internal const int CacheCapacity = 40;
```

Comment:
I did not realize the number of keys that will be held by the cache is fixed, sorry. If that is the case though, then there is no concern of unlimited growth and the fixed capacity is not necessary IMO.